ipc/cli/src/commands/subnet/create.rs

```diff
@@ -70,6 +64,35 @@ impl CreateSubnet {
     }
 }
 
+fn parse_supply_source(arguments: &CreateSubnetArgs) -> anyhow::Result<Asset> {
+    let token_address = if let Some(addr) = &arguments.supply_source_address {
+        Some(require_fil_addr_from_str(addr)?)
+    } else {
+        None
+    };
+    Ok(Asset {
+        kind: arguments.supply_source_kind,
+        token_address,
+    })
+}
+
+fn parse_collateral_source(arguments: &CreateSubnetArgs) -> anyhow::Result<Asset> {
+    let Some(ref kind) = arguments.collateral_source_kind else {
+        return Ok(Asset::default());
+    };
+
+    let token_address = if let Some(addr) = &arguments.collateral_source_address {
```

Comment:
Do we want to enforce that when the token is ERC20 the address should be set?

Reply:
I think contract should have this tested. I dont see much of a point in checking, simply that we donno if the address is indeed correct.

Reply:
True, but we can also fail fast with nicer error here.